---
title: "The depmap data"
author: "Theo Killian"
date: "February 19, 2019"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteEngine{knitr::knitr}
  %\VignetteIndexEntry{depmap}
  %\usepackage[utf8]{inputenc}
---

```{r, echo = FALSE, results = 'asis'}
suppressPackageStartupMessages(library("dplyr"))
knitr::opts_chunk$set(collapse = TRUE, comment = "#>")
```

# [depmap](https://github.com/tfkillian/depmap/)

This vignette illustrates the steps in loading of the data found in the
[depmap](https://github.com/tfkillian/depmap/) package.

## Introduction {#sec:intro}

The [depmap](https://github.com/tfkillian/depmap/) package aims to provide a
reproducible research framework to cancer dependency data described by
[Tsherniak, Aviad, et al. "Defining a cancer dependency map." Cell 170.3 (2017):
564-576.](https://www.ncbi.nlm.nih.gov/pubmed/28753430). The data found in the
[depmap](https://github.com/tfkillian/depmap/) package has been formatted to
facilitate the use of common R packages such as `dplyr` and `ggplot2`. We hope
that this package will allow researchers to more easily mine, explore and
visually illustrate dependency data taken from the Depmap cancer genomic
dependency study.

## Preliminaries

To install [depmap](https://github.com/tfkillian/depmap/), the
[BiocManager](https://cran.r-project.org/web/packages/BiocManager/index.html)
Bioconductor Project Package Manager is required. If
[BiocManager](https://cran.r-project.org/web/packages/BiocManager/index.html) is
not already installed, it will need to be done so beforehand. Type (within R)
install.packages("BiocManager") (This needs to be done just once.)

```{r depmap_bioconductor_installation, eval=FALSE}
install.packages("BiocManager")
BiocManager::install("UCLouvain-CBIO/depmap")
```

## Data Import {#sec:data}

The [depmap](https://github.com/tfkillian/depmap/) package currently contains
seven datasets in *.rda* format. These datasets can be loaded by using *data()*
after this package has been installed and loaded in RStudio with
*library("depmap")*. The data found in this R package has been converted from
wide format to long format, although none of the values taken from the original
datasets has been changed. The code related to the data transformation is
can be found in the make_data.R file found in ./depmap/inst/scripts

```{r load_data, echo = FALSE}
data("rnai", package = "depmap")
data("crispr", package = "depmap")
data("copyNumber", package = "depmap")
data("RPPA", package = "depmap")
data("TPM", package = "depmap")
data("metadata", package = "depmap")
data("mutationCalls", package = "depmap")
```

### `rnai`

The `rnai` dataset contains the combined genetic dependency data for RNAi -
induced gene knockdown for select genes and cancer cell lines. This data
corresponds to the `D2_combined_genetic_dependency_scores.csv` file found in the
`r depmap::depmap_release()` depmap release and includes `r length(unique(rnai$gene))`
genes, `r length(unique(rnai$cell_line))` cell lines, 30 primary diseases and 31
lineages.

```{r show_rnai}
rnai
```

### `crispr`

The `crispr` dataset contains the (batch corrected CERES inferred gene effect)
CRISPR-Cas9 knockout data of select genes and cancer cell lines. This data
corresponds to the `gene_effect_corrected.csv` file from the `r depmap::depmap_release()`
depmap release. Data from this dataset includes `r length(unique(crispr$gene_name))`
genes, `r length(unique(crispr$cell_line))` cell lines, 26 primary diseases, 28
lineages.

```{r show_crispr}
crispr
```

### `copyNumber`

The `copyNumber` dataset contains the WES copy number data, relating to the
numerical log-fold copy number change measured against the baseline copy number
of select genes and cell lines. This dataset corresponds to the
`public_19Q1_gene_cn.csv` from the `r depmap::depmap_release()` depmap release.
This dataset includes `r length(unique(copyNumber$gene_name))` genes,
`r length(unique(crispr$cell_line))` cell lines, 38 primary diseases and 33
lineages.

```{r show_copynumber}
copyNumber
```

### `RPPA`

The `RPPA` dataset contains the CCLE Reverse Phase Protein Array (RPPA) data 
which corresponds to the `CCLE_RPPA_20180123.csv` file from the `r depmap::depmap_release()`
depmap release. This dataset includes `r length(unique(RPPA$gene_name))` genes,
`r length(unique(RPPA$cell_line))` cell lines, 28 primary diseases, 28 lineages.

```{r show_rppa}
RPPA
```

### `TPM`

The `TPM` dataset contains the CCLE RNAseq gene expression data. This shows
expression data only for protein coding genes (using scale log2(TPM+1)). This
data corresponds to the `CCLE_depMap_19Q1_TPM.csv` file from the `r depmap::depmap_release()` 
depmap release. This dataset includes `r length(unique(TPM$gene_name))` genes,
`r length(unique(TPM$cell_line))` cell lines, 33 primary Diseases, 32 lineages.

```{r show_tpm}
TPM
```

### `metadata`

The `metadata` dataset contains the metadata about all of the cancer cell lines.
It corresponds to the `depmap_19Q1_cell_lines.csv` file found in the `r depmap::depmap_release()`
depmap release. This dataset includes `r length(unique(metadata$gene_name))`
genes, `r length(unique(metadata$cell_line))` cell lines, 38 primary diseases
and 33 lineages.

```{r show_metadata}
metadata
```

### `mutationCalls`

The `mutationCalls` dataset contains all merged mutation calls (coding region,
germline filtered) found in the depmap dependency study. This dataset
corresponds with the `depmap_19Q1_mutation_calls.csv` file found in the `r depmap::depmap_release()`
depmap release and includes `r length(unique(mutationCalls$gene_name))` genes,
`r length(unique(mutationCalls$depmap_id))` cell lines, 37 primary diseases and
33 lineages.

```{r show_mutationcalls}
mutationCalls
```

### Downloading [Broad Institute](https://depmap.org/portal/download/) data

If desired, the original wide form data from which the long form datasets found
in the [depmap](https://github.com/tfkillian/depmap/) package were derived can
be downloaded from the [Broad Institute](https://depmap.org/portal/download/)
website. The instructions on how to download these files and how the data was
transformed and loaded into the [depmap](https://github.com/tfkillian/depmap/)
package can be found in the make_data.R file found in ./inst/scripts. (It should
be noted that the original uncompressed *.csv* files are 1.5GB in total and
take a moderate amount of time to download.)
