<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>mutationCalls_22Q2 — mutationCalls • depmap</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="mutationCalls_22Q2 — mutationCalls"><meta name="description" content="The `mutationCalls` dataset contains merged the 22Q2 mutation calls (for
coding region, germline filtered) and includes data from 18784 genes, 1771
cell lines, 33 primary diseases and 30 lineages. This dataset can be
considered the metadata data set for mutations and does not contain any
dependency data. This dataset can be loaded into the R environment with the
`depmap_mutationCalls` function."><meta property="og:description" content="The `mutationCalls` dataset contains merged the 22Q2 mutation calls (for
coding region, germline filtered) and includes data from 18784 genes, 1771
cell lines, 33 primary diseases and 30 lineages. This dataset can be
considered the metadata data set for mutations and does not contain any
dependency data. This dataset can be loaded into the R environment with the
`depmap_mutationCalls` function."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">depmap</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.19.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/depmap.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/using_depmap.html">Using the depmap data</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/UCLouvain-CBIO/depmap/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>mutationCalls_22Q2</h1>
      <small class="dont-index">Source: <a href="https://github.com/UCLouvain-CBIO/depmap/blob/master/R/mutationCalls.R" class="external-link"><code>R/mutationCalls.R</code></a></small>
      <div class="d-none name"><code>mutationCalls.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>The `mutationCalls` dataset contains merged the 22Q2 mutation calls (for
coding region, germline filtered) and includes data from 18784 genes, 1771
cell lines, 33 primary diseases and 30 lineages. This dataset can be
considered the metadata data set for mutations and does not contain any
dependency data. This dataset can be loaded into the R environment with the
`depmap_mutationCalls` function.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="va">mutationCalls</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="format">Format<a class="anchor" aria-label="anchor" href="#format"></a></h2>
    <p>A data frame with 1235466 rows and 32 variables:</p><dl><dt>depmap_id</dt>
<dd><p>depmap_id</p></dd>

  <dt>gene_name</dt>
<dd><p>Hugo Symbol denotes a unique and meaningful name for each
  gene (e.g. SAP25)</p></dd>

  <dt>entrez_id</dt>
<dd><p>Gene ID for NCBI Entrez gene database, (e.g. 100316904)</p></dd>

  <dt>ncbi_build</dt>
<dd><p>NCBI Build (i.e. reference genome)</p></dd>

  <dt>chromosome</dt>
<dd><p>Chromosome</p></dd>

  <dt>start_pos</dt>
<dd><p>Gene start position</p></dd>

  <dt>end_pos</dt>
<dd><p>Gene end position</p></dd>

  <dt>strand</dt>
<dd><p>Strand location of gene</p></dd>

  <dt>var_class</dt>
<dd><p>Variant Classification</p></dd>

  <dt>var_type</dt>
<dd><p>Variant Type</p></dd>

  <dt>ref_allele</dt>
<dd><p>Reference Allele</p></dd>

  <dt>alt_allele</dt>
<dd><p>Tumor Seq Allele1</p></dd>

  <dt>dbSNP_RS</dt>
<dd><p>Single Nucleotide Polymorphism Database (dbSNP) reference
   cluster</p></dd>

  <dt>dbSNP_val_status</dt>
<dd><p>dbSNP Val Status</p></dd>

  <dt>genome_change</dt>
<dd><p>Genome Change</p></dd>

  <dt>annotation_transcript</dt>
<dd><p>Annotation Transcript</p></dd>

  <dt>cDNA_change</dt>
<dd><p>change in cDNA</p></dd>

  <dt>codon_change</dt>
<dd><p>Codon_Change</p></dd>

  <dt>protein_change</dt>
<dd><p>Protein_Change</p></dd>

  <dt>is_deleterious</dt>
<dd><p>Status of gene knockout on cell lineage</p></dd>

  <dt>is_tcga_hotspot</dt>
<dd><p>isTCGAhotspot</p></dd>

  <dt>tcga_hsCnt</dt>
<dd><p>TCGAhsCnt</p></dd>

  <dt>is_cosmic_hotspot</dt>
<dd><p>isCOSMIChotspot</p></dd>

  <dt>cosmic_hsCnt</dt>
<dd><p>COSMIChsCnt</p></dd>

  <dt>ExAC_AF</dt>
<dd><p>ExAC_AF</p></dd>

  <dt>CGA_WES_AC</dt>
<dd><p>CGA_WES_AC</p></dd>

  <dt>sanger_WES_AC</dt>
<dd><p>SangerWES_AC</p></dd>

  <dt>RNAseq_AC</dt>
<dd><p>RNAseq_AC</p></dd>

  <dt>HC_AC</dt>
<dd><p>HC_AC</p></dd>

  <dt>RD_AC</dt>
<dd><p>RD_AC</p></dd>

  <dt>WGS_AC</dt>
<dd><p>WGS_AC</p></dd>

  <dt>var_annotation</dt>
<dd><p>Variant_annotation</p></dd>


</dl></div>
    <div class="section level2">
    <h2 id="source">Source<a class="anchor" aria-label="anchor" href="#source"></a></h2>
    <p>DepMap, Broad Institute: https://depmap.org/portal/download/</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>This data represents the `CCLE_mutations.csv` file taken from the
22Q2 [Broad Institute](https://depmap.org/portal/download/) cancer
depenedency study. The derived dataset found in the `depmap` package features
the addition of a foreign key `depmap_id` found in the first column of this
dataset, which was added from the `metadata` dataset. This dataset has been
converted to a long format tibble. Variables names from the original dataset
were converted to lower case, put in snake case, and abbreviated where
feasible.</p>
    </div>
    <div class="section level2">
    <h2 id="change-log">Change log<a class="anchor" aria-label="anchor" href="#change-log"></a></h2>



<p>- 19Q1: Initial dataset for package consisted of dataframe with
  1243145 rows and 35 variables representing 18755 genes, 1601 cell
  lines, 37 primary diseases and 33 lineages.</p>
<p>- 19Q2: adds 30 cell lines, 1 primary disease and 1 lineage. This
  version has different columns than the previous version: the
  variable "VA_WES_AC" is no longer present in this dataset. Some
  minor alterations to the original file were made. The first
  column of the original dataset, (ID, Sample number) was removed,
  as this column was only the row number and did not serve any
  unique identifying purpose.</p>
<p>- 19Q3: adds 1 gene, 25 cell lines and removes 1 primary disease.</p>
<p>- 19Q4: adds 1 gene, 10 cell lines, 0 primary diseases and 2
  lineages.</p>
<p>- 20Q1: adds 4 genes, 31 cell lines, 1 lineage.</p>
<p>- 20Q2: adds 44 cell lines, 1 lineage.</p>
<p>- 20Q3: no change.</p>
<p>- 20Q4: removes 13 genes, adds 8 cell lines and 1 lineage. Columns
 `tumor_sample_barcode` and `sanger_recalib_WES_AC` were removed.</p>
<p>- 21Q1: removes 11 genes and 2 cell lines.</p>
<p>- 21Q2: removes 1 genes and adds 3 cell lines.</p>
<p>- 21Q3: removes 3 genes, 4 cell lines and 1 lineage.</p>
<p>- 21Q3: removes 3 genes, 4 cell lines and 1 lineage.</p>
<p>- 21Q4: adds 9 cell lines.</p>
<p>- 22Q1: adds 4 cell lines and 1 lineage. The variable `tumor_seq_allele1`
  was renamed `alt_allele`.</p>
<p>- 22Q2: adds 12 cell lines and removes 2 primary diseases and 8
  lineages.</p>
    </div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>Tsherniak, A., Vazquez, F., Montgomery, P. G., Weir, B. A.,
Kryukov, G., Cowley, G. S., ... &amp; Meyers, R. M. (2017). Defining a cancer
dependency map. Cell, 170(3), 564-576.</p>
<p>DepMap, Broad (2019): DepMap Achilles 19Q1
Public. https://figshare.com/articles/DepMap_Achilles_19Q1_Public/7655150</p>
<p>Robin M. Meyers, Jordan G. Bryan, James M. McFarland, Barbara
A. Weir, ...  David E. Root, William C. Hahn, Aviad
Tsherniak. Computational correction of copy number effect improves
specificity of CRISPR-Cas9 essentiality screens in cancer
cells. Nature Genetics 2017 October 49:1779–1784.</p>
<p>Mahmoud Ghandi, Franklin W. Huang, Judit Jané-Valbuena, Gregory V. Kryukov,
... Todd R. Golub, Levi A. Garraway &amp; William R. Sellers. 2019. Next-
generation characterization of the Cancer Cell Line Encyclopedia. Nature 569,
503–508 (2019).</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span> <span class="co"># \dontrun{</span></span></span>
<span class="r-in"><span><span class="fu">depmap_mutationCalls</span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span> <span class="co"># }</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Laurent Gatto, Theo Killian.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.9000.</p>
</div>

    </footer></div>





  </body></html>

