<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>TPM_22Q2 — TPM • depmap</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="TPM_22Q2 — TPM"><meta name="description" content='The `TPM` dataset contains the 22Q2 CCLE "Transcript Per Million" RNAseq gene
expression data for protein coding genes. This dataset includes data from
19221 genes, 1406 cell lines, 33 primary diseases and 30 lineages. The
columns of `TPM` are: `depmap_id`, a foreign key corresponding to the cancer
cell lineage, `cell_line` the common CCLE name of the cancer cell lines,
`gene` containing both the HUGO gene name of the knockdown gene along with
ensembl ID#, `gene_name` containing the HUGO gene name and `ensembl_id`
containing only the ensembl ID# and `rna_expression` which contains the
numerical protein coding gene expression change at scale (log2(TPM+1)). This
dataset can be loaded into R environment with the `depmap_TPM` function.'><meta property="og:description" content='The `TPM` dataset contains the 22Q2 CCLE "Transcript Per Million" RNAseq gene
expression data for protein coding genes. This dataset includes data from
19221 genes, 1406 cell lines, 33 primary diseases and 30 lineages. The
columns of `TPM` are: `depmap_id`, a foreign key corresponding to the cancer
cell lineage, `cell_line` the common CCLE name of the cancer cell lines,
`gene` containing both the HUGO gene name of the knockdown gene along with
ensembl ID#, `gene_name` containing the HUGO gene name and `ensembl_id`
containing only the ensembl ID# and `rna_expression` which contains the
numerical protein coding gene expression change at scale (log2(TPM+1)). This
dataset can be loaded into R environment with the `depmap_TPM` function.'></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">depmap</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.19.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/depmap.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/using_depmap.html">Using the depmap data</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/UCLouvain-CBIO/depmap/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>TPM_22Q2</h1>
      <small class="dont-index">Source: <a href="https://github.com/UCLouvain-CBIO/depmap/blob/master/R/TPM.R" class="external-link"><code>R/TPM.R</code></a></small>
      <div class="d-none name"><code>TPM.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>The `TPM` dataset contains the 22Q2 CCLE "Transcript Per Million" RNAseq gene
expression data for protein coding genes. This dataset includes data from
19221 genes, 1406 cell lines, 33 primary diseases and 30 lineages. The
columns of `TPM` are: `depmap_id`, a foreign key corresponding to the cancer
cell lineage, `cell_line` the common CCLE name of the cancer cell lines,
`gene` containing both the HUGO gene name of the knockdown gene along with
ensembl ID#, `gene_name` containing the HUGO gene name and `ensembl_id`
containing only the ensembl ID# and `rna_expression` which contains the
numerical protein coding gene expression change at scale (log2(TPM+1)). This
dataset can be loaded into R environment with the `depmap_TPM` function.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="va">TPM</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="format">Format<a class="anchor" aria-label="anchor" href="#format"></a></h2>
    <p>A data frame with 27024726 rows (cell lines) and 6 variables:</p><dl><dt>depmap_id</dt>
<dd><p>Cell line foreign key (i.e. "ACH-000956")</p></dd>

     <dt>cell_line</dt>
<dd><p>Name of cancer cell line (i.e. "22RV1_PROSTATE")</p></dd>

     <dt>gene</dt>
<dd><p>HUGO symbol and Ensembl ID (e.g. TSPAN6 (ENSG00000000003))</p></dd>

     <dt>gene_name</dt>
<dd><p>HUGO symbol (e.g. "TSPAN6")</p></dd>

     <dt>ensembl_id</dt>
<dd><p>Ensembl ID (e.g. ENSG00000044574)</p></dd>

     <dt>rna_expression</dt>
<dd><p>Log fold (log2(TPM+1)) protein expression change</p></dd>


</dl></div>
    <div class="section level2">
    <h2 id="source">Source<a class="anchor" aria-label="anchor" href="#source"></a></h2>
    <p>DepMap, Broad Institute: https://depmap.org/portal/download/</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>This data originates from the `CCLE_expression.csv` file taken
from the 22Q2 [Broad Institute](https://depmap.org/portal/download/) cancer
depenedency study. The derived dataset found in the `depmap` package features
the addition of a foreign key `depmap_id` found in the first column of this
dataset, which was added from the `metadata` dataset. This dataset has been
converted to a long format tibble. Variables names from the original dataset
were converted to lower case, put in snake case, and abbreviated where
feasible.</p>
    </div>
    <div class="section level2">
    <h2 id="change-log">Change log<a class="anchor" aria-label="anchor" href="#change-log"></a></h2>



<p>- 19Q1: Initial dataset consisted of a data frame with 67360300 rows (cell
lines) and 6 variables representing 57820 genes, 1165 cell lines, 33
primary diseases, 32 lineages.</p>
<p>- 19Q2: removes 1618 genes, adds 36 cell lines, removes one primary disease
and adds 1 lineage</p>
<p>- 19Q3: removes 37058 genes, adds 9 cell lines, removes 2 primary diseases.
Now a 23164240 by 6 dataframe.</p>
<p>- 19Q4: 0 genes, 39 cancer cell lines, 0 primary diseases and 1 lineage</p>
<p>- 20Q1: adds 31 cell lines</p>
<p>- 20Q2: adds 34 cell lines. `expression` changed to `rna_expression`</p>
<p>- 20Q3: adds 1 cell line</p>
<p>- 20Q4: removes 38 genes and 71 cell lines, addes 1 primary disease and 3
 lineages</p>
<p>- 21Q1: removes 5 genes</p>
<p>- 21Q2: adds 3 cell lines. Additionally, a bug was fixed where the Entrez ID
appeared as Ensembl_ID. This was changed for all previous versions of this
dataset from 19Q3 to 21Q2</p>
<p>- 21Q3: removes 2 cell lines</p>
<p>- 21Q4: adds 12 cell lines</p>
<p>- 22Q1: adds 4 cell lines and 1 lineage</p>
<p>- 22Q2: adds 44 genes, 13 cell lines and removes 8 lineages</p>
    </div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>Tsherniak, A., Vazquez, F., Montgomery, P. G., Weir, B. A.,
Kryukov, G., Cowley, G. S., ... &amp; Meyers, R. M. (2017). Defining a cancer
dependency map. Cell, 170(3), 564-576.</p>
<p>DepMap, Broad (2019): DepMap Achilles 19Q1
Public. https://figshare.com/articles/DepMap_Achilles_19Q1_Public/7655150.
Fileset.</p>
<p>Robin M. Meyers, Jordan G. Bryan, James M. McFarland, Barbara A. Weir, ...
David E. Root, William C. Hahn, Aviad Tsherniak. Computational correction of
copy number effect improves specificity of CRISPR-Cas9 essentiality screens
in cancer cells. Nature Genetics 2017 October 49:1779–1784.</p>
<p>Mahmoud Ghandi, Franklin W. Huang, Judit Jané-Valbuena, Gregory V. Kryukov,
... Todd R. Golub, Levi A. Garraway &amp; William R. Sellers. 2019. Next-
generation characterization of the Cancer Cell Line Encyclopedia. Nature 569,
503–508 (2019).</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span> <span class="co"># \dontrun{</span></span></span>
<span class="r-in"><span><span class="fu">depmap_TPM</span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span> <span class="co"># }</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Laurent Gatto, Theo Killian.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.9000.</p>
</div>

    </footer></div>





  </body></html>

