<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>The depmap data • depmap</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="The depmap data">
<meta property="og:description" content="depmap">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">depmap</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.17.5</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/depmap.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/using_depmap.html">Using the depmap data</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>The depmap data</h1>
                        <h4 data-toc-skip class="author">Theo
Killian</h4>
                                    <h4 data-toc-skip class="author">Laurent
Gatto</h4>
            <address class="author_afil">
      Computational Biology, UCLouvain<br><h4 data-toc-skip class="date">2024-03-21</h4>
      
      
      <div class="hidden name"><code>depmap.Rmd</code></div>

    </address>
</div>

    
    
<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  document.querySelector("h1").className = "title";
});
</script><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if (links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script><div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The <code>depmap</code> package aims to provide a reproducible
research framework to cancer dependency data described by <a href="https://www.ncbi.nlm.nih.gov/pubmed/28753430" class="external-link">Tsherniak, Aviad, et
al. “Defining a cancer dependency map.” Cell 170.3 (2017): 564-576.</a>.
The data found in the <a href="https://bioconductor.org/packages/devel/data/experiment/html/depmap.html" class="external-link">depmap</a>
package has been formatted to facilitate the use of common R packages
such as <code>dplyr</code> and <code>ggplot2</code>. We hope that this
package will allow researchers to more easily mine, explore and visually
illustrate dependency data taken from the Depmap cancer genomic
dependency study.</p>
</div>
<div class="section level2">
<h2 id="installation-instructions">Installation instructions<a class="anchor" aria-label="anchor" href="#installation-instructions"></a>
</h2>
<p>To install <a href="https://bioconductor.org/packages/devel/data/experiment/html/depmap.html" class="external-link">depmap</a>,
the <a href="https://cran.r-project.org/web/packages/BiocManager/index.html" class="external-link">BiocManager</a>
Bioconductor Project Package Manager is required. If <a href="https://cran.r-project.org/web/packages/BiocManager/index.html" class="external-link">BiocManager</a>
is not already installed, it will need to be done so beforehand. Type
(within R) install.packages(“BiocManager”) (This needs to be done just
once.)</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"depmap"</span><span class="op">)</span></span></code></pre></div>
<p>The <code>depmap</code> package fully depends on the
<code>ExperimentHub</code> Bioconductor package, which allows the data
accessed in this package to be stored and retrieved from the cloud.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"depmap"</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"ExperimentHub"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="tidy-depmap-data">Tidy depmap data<a class="anchor" aria-label="anchor" href="#tidy-depmap-data"></a>
</h2>
<p>The <a href="https://bioconductor.org/packages/devel/data/experiment/html/depmap.html" class="external-link">depmap</a>
package currently contains eight datasets available through
<code>ExperimentHub</code>.</p>
<p>The data found in this R package has been converted from a “wide”
format <code>.csv</code> file to “long” format .rda file. None of the
values taken from the original datasets have been changed, although the
columns have been re-arranged. Descriptions of the changes made are
described under the <code>Details</code> section after querying the
relevant dataset.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## create ExperimentHub query object</span></span>
<span><span class="va">eh</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ExperimentHub/man/ExperimentHub-class.html" class="external-link">ExperimentHub</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">query</a></span><span class="op">(</span><span class="va">eh</span>, <span class="st">"depmap"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## ExperimentHub with 82 records</span></span>
<span><span class="co">## # snapshotDate(): 2024-03-11</span></span>
<span><span class="co">## # $dataprovider: Broad Institute</span></span>
<span><span class="co">## # $species: Homo sapiens</span></span>
<span><span class="co">## # $rdataclass: tibble</span></span>
<span><span class="co">## # additional mcols(): taxonomyid, genome, description,</span></span>
<span><span class="co">## #   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,</span></span>
<span><span class="co">## #   rdatapath, sourceurl, sourcetype </span></span>
<span><span class="co">## # retrieve records with, e.g., 'object[["EH2260"]]' </span></span>
<span><span class="co">## </span></span>
<span><span class="co">##            title             </span></span>
<span><span class="co">##   EH2260 | rnai_19Q1         </span></span>
<span><span class="co">##   EH2261 | crispr_19Q1       </span></span>
<span><span class="co">##   EH2262 | copyNumber_19Q1   </span></span>
<span><span class="co">##   EH2263 | RPPA_19Q1         </span></span>
<span><span class="co">##   EH2264 | TPM_19Q1          </span></span>
<span><span class="co">##   ...      ...               </span></span>
<span><span class="co">##   EH7555 | copyNumber_22Q2   </span></span>
<span><span class="co">##   EH7556 | TPM_22Q2          </span></span>
<span><span class="co">##   EH7557 | mutationCalls_22Q2</span></span>
<span><span class="co">##   EH7558 | metadata_22Q2     </span></span>
<span><span class="co">##   EH7559 | achilles_22Q2</span></span></code></pre>
<p>Each dataset has a <code>ExperimentHub</code> accession number,
(e.g. <em>EH2260</em> refers to the <code>rnai</code> dataset from the
19Q1 release).</p>
<div class="section level3">
<h3 id="rna-inference-knockout-data">RNA inference knockout data<a class="anchor" aria-label="anchor" href="#rna-inference-knockout-data"></a>
</h3>
<p>The <code>rnai</code> dataset contains the combined genetic
dependency data for RNAi - induced gene knockdown for select genes and
cancer cell lines. This data corresponds to the
<code>D2_combined_genetic_dependency_scores.csv</code> file.</p>
<p>Specific <code>rnai</code> datasets can be accessed, such as
<code>rnai_19Q1</code> by EH number.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">eh</span><span class="op">[[</span><span class="st">"EH2260"</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<p>The most recent <code>rnai</code> dataset can be automatically loaded
into R by using the <code>depmap_rnai</code> function.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">depmap</span><span class="fu">::</span><span class="fu"><a href="../reference/rnai.html">depmap_rnai</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## see ?depmap and browseVignettes('depmap') for documentation</span></span></code></pre>
<pre><code><span><span class="co">## loading from cache</span></span></code></pre>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 12,324,008 × 6</span></span></span>
<span><span class="co">##    gene                cell_line        dependency entrez_id gene_name depmap_id</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>               <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> A1BG (1)            127399_SOFT_TIS…     <span style="color: #BB0000;">NA</span>             1 A1BG      ACH-0012…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> NAT2 (10)           127399_SOFT_TIS…     <span style="color: #BB0000;">NA</span>            10 NAT2      ACH-0012…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> ADA (100)           127399_SOFT_TIS…     <span style="color: #BB0000;">NA</span>           100 ADA       ACH-0012…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> CDH2 (1000)         127399_SOFT_TIS…     -<span style="color: #BB0000;">0.195</span>      <span style="text-decoration: underline;">1</span>000 CDH2      ACH-0012…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> AKT3 (10000)        127399_SOFT_TIS…     -<span style="color: #BB0000;">0.256</span>     <span style="text-decoration: underline;">10</span>000 AKT3      ACH-0012…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> MED6 (10001)        127399_SOFT_TIS…     -<span style="color: #BB0000;">0.174</span>     <span style="text-decoration: underline;">10</span>001 MED6      ACH-0012…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> NR2E3 (10002)       127399_SOFT_TIS…     -<span style="color: #BB0000;">0.140</span>     <span style="text-decoration: underline;">10</span>002 NR2E3     ACH-0012…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> NAALAD2 (10003)     127399_SOFT_TIS…     <span style="color: #BB0000;">NA</span>         <span style="text-decoration: underline;">10</span>003 NAALAD2   ACH-0012…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> DUXB (100033411)    127399_SOFT_TIS…     <span style="color: #BB0000;">NA</span>     100<span style="text-decoration: underline;">033</span>411 DUXB      ACH-0012…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> PDZK1P1 (100034743) 127399_SOFT_TIS…     <span style="color: #BB0000;">NA</span>     100<span style="text-decoration: underline;">034</span>743 PDZK1P1   ACH-0012…</span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 12,323,998 more rows</span></span></span></code></pre>
</div>
<div class="section level3">
<h3 id="crispr-cas9-knockout-data">CRISPR-Cas9 knockout data<a class="anchor" aria-label="anchor" href="#crispr-cas9-knockout-data"></a>
</h3>
<p>The <code>crispr</code> dataset contains the (batch corrected CERES
inferred gene effect) CRISPR-Cas9 knockout data of select genes and
cancer cell lines. This data corresponds to the
<code>gene_effect_corrected.csv</code> file.</p>
<p>Specific <code>crispr</code> datasets can be accessed, such as
<code>crispr_19Q1</code> by EH number.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">eh</span><span class="op">[[</span><span class="st">"EH2261"</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<p>The most recent <code>crispr</code> dataset can be automatically
loaded into R by using the <code>depmap_crispr</code> function.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">depmap</span><span class="fu">::</span><span class="fu"><a href="../reference/crispr.html">depmap_crispr</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## see ?depmap and browseVignettes('depmap') for documentation</span></span></code></pre>
<pre><code><span><span class="co">## loading from cache</span></span></code></pre>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 18,881,196 × 6</span></span></span>
<span><span class="co">##    depmap_id  gene     dependency entrez_id gene_name cell_line                 </span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> ACH-000001 A1BG (1)    -<span style="color: #BB0000;">0.135</span>          1 A1BG      NIHOVCAR3_OVARY           </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> ACH-000004 A1BG (1)     0.081<span style="text-decoration: underline;">9</span>         1 A1BG      HEL_HAEMATOPOIETIC_AND_LY…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> ACH-000005 A1BG (1)    -<span style="color: #BB0000;">0.094</span><span style="color: #BB0000; text-decoration: underline;">2</span>         1 A1BG      HEL9217_HAEMATOPOIETIC_AN…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> ACH-000007 A1BG (1)    -<span style="color: #BB0000;">0.011</span><span style="color: #BB0000; text-decoration: underline;">5</span>         1 A1BG      LS513_LARGE_INTESTINE     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> ACH-000009 A1BG (1)    -<span style="color: #BB0000;">0.050</span><span style="color: #BB0000; text-decoration: underline;">8</span>         1 A1BG      C2BBE1_LARGE_INTESTINE    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> ACH-000011 A1BG (1)     0.091<span style="text-decoration: underline;">8</span>         1 A1BG      253J_URINARY_TRACT        </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> ACH-000012 A1BG (1)    -<span style="color: #BB0000;">0.147</span>          1 A1BG      HCC827_LUNG               </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> ACH-000013 A1BG (1)    -<span style="color: #BB0000;">0.059</span><span style="color: #BB0000; text-decoration: underline;">2</span>         1 A1BG      ONCODG1_OVARY             </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> ACH-000014 A1BG (1)    -<span style="color: #BB0000;">0.034</span><span style="color: #BB0000; text-decoration: underline;">8</span>         1 A1BG      HS294T_SKIN               </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> ACH-000015 A1BG (1)    -<span style="color: #BB0000;">0.204</span>          1 A1BG      NCIH1581_LUNG             </span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 18,881,186 more rows</span></span></span></code></pre>
</div>
<div class="section level3">
<h3 id="wes-copy-number-data">WES copy number data<a class="anchor" aria-label="anchor" href="#wes-copy-number-data"></a>
</h3>
<p>The <code>copyNumber</code> dataset contains the WES copy number
data, relating to the numerical log-fold copy number change measured
against the baseline copy number of select genes and cell lines. This
dataset corresponds to the <code>public_19Q1_gene_cn.csv</code></p>
<p>Specific <code>copyNumber</code> datasets can be accessed, such as
<code>copyNumber_19Q1</code> by EH number.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">eh</span><span class="op">[[</span><span class="st">"EH2262"</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<p>The most recent <code>copyNumber</code> dataset can be automatically
loaded into R by using the <code>depmap_copyNumber</code> function.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">depmap</span><span class="fu">::</span><span class="fu"><a href="../reference/copyNumber.html">depmap_copyNumber</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## see ?depmap and browseVignettes('depmap') for documentation</span></span></code></pre>
<pre><code><span><span class="co">## loading from cache</span></span></code></pre>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 44,799,888 × 6</span></span></span>
<span><span class="co">##    depmap_id  gene            log_copy_number entrez_id gene_name cell_line     </span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> ACH-000267 DDX11L1 (84771)           1.15      <span style="text-decoration: underline;">84</span>771 DDX11L1   HDLM2_HAEMATO…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> ACH-001408 DDX11L1 (84771)           1.04      <span style="text-decoration: underline;">84</span>771 DDX11L1   UMUC14_URINAR…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> ACH-000617 DDX11L1 (84771)           0.762     <span style="text-decoration: underline;">84</span>771 DDX11L1   OVCAR4_OVARY  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> ACH-002123 DDX11L1 (84771)           1.14      <span style="text-decoration: underline;">84</span>771 DDX11L1   H2369_PLEURA  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> ACH-000519 DDX11L1 (84771)           1.01      <span style="text-decoration: underline;">84</span>771 DDX11L1   PEER_HAEMATOP…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> ACH-000750 DDX11L1 (84771)           0.711     <span style="text-decoration: underline;">84</span>771 DDX11L1   LOXIMVI_SKIN  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> ACH-000544 DDX11L1 (84771)           0.981     <span style="text-decoration: underline;">84</span>771 DDX11L1   OE21_OESOPHAG…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> ACH-001214 DDX11L1 (84771)           1.05      <span style="text-decoration: underline;">84</span>771 DDX11L1   U138MG_CENTRA…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> ACH-002223 DDX11L1 (84771)           0.630     <span style="text-decoration: underline;">84</span>771 DDX11L1   D245MG_CENTRA…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> ACH-000713 DDX11L1 (84771)           0.823     <span style="text-decoration: underline;">84</span>771 DDX11L1   CAOV3_OVARY   </span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 44,799,878 more rows</span></span></span></code></pre>
</div>
<div class="section level3">
<h3 id="ccle-reverse-phase-protein-array-data">CCLE Reverse Phase Protein Array data<a class="anchor" aria-label="anchor" href="#ccle-reverse-phase-protein-array-data"></a>
</h3>
<p>The <code>RPPA</code> dataset contains the CCLE Reverse Phase Protein
Array (RPPA) data which corresponds to the
<code>CCLE_RPPA_20180123.csv</code> file.</p>
<p>Specific <code>RPPA</code> datasets can be accessed, such as
<code>RPPA_19Q1</code> by EH number.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">eh</span><span class="op">[[</span><span class="st">"EH2263"</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<p>The most recent <code>RPPA</code> dataset can be automatically loaded
into R by using the <code>depmap_RPPA</code> function.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">depmap</span><span class="fu">::</span><span class="fu"><a href="../reference/RPPA.html">depmap_RPPA</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## see ?depmap and browseVignettes('depmap') for documentation</span></span></code></pre>
<pre><code><span><span class="co">## loading from cache</span></span></code></pre>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 192,386 × 4</span></span></span>
<span><span class="co">##    cell_line                                 antibody    expression depmap_id </span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> DMS53_LUNG                                14-3-3_beta    -<span style="color: #BB0000;">0.105</span>  ACH-000698</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> SW1116_LARGE_INTESTINE                    14-3-3_beta     0.359  ACH-000489</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> NCIH1694_LUNG                             14-3-3_beta     0.028<span style="text-decoration: underline;">7</span> ACH-000431</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> P3HR1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  14-3-3_beta     0.120  ACH-000707</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> HUT78_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  14-3-3_beta    -<span style="color: #BB0000;">0.269</span>  ACH-000509</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> UMUC3_URINARY_TRACT                       14-3-3_beta    -<span style="color: #BB0000;">0.171</span>  ACH-000522</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> HOS_BONE                                  14-3-3_beta    -<span style="color: #BB0000;">0.025</span><span style="color: #BB0000; text-decoration: underline;">3</span> ACH-000613</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> HUNS1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  14-3-3_beta    -<span style="color: #BB0000;">0.170</span>  ACH-000829</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> AML193_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE 14-3-3_beta     0.081<span style="text-decoration: underline;">9</span> ACH-000557</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> RVH421_SKIN                               14-3-3_beta     0.222  ACH-000614</span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 192,376 more rows</span></span></span></code></pre>
</div>
<div class="section level3">
<h3 id="ccle-rnaseq-gene-expression-data">CCLE RNAseq gene expression data<a class="anchor" aria-label="anchor" href="#ccle-rnaseq-gene-expression-data"></a>
</h3>
<p>The <code>TPM</code> dataset contains the CCLE RNAseq gene expression
data. This shows expression data only for protein coding genes (using
scale log2(TPM+1)). This data corresponds to the
<code>CCLE_depMap_19Q1_TPM.csv</code> file.</p>
<p>Specific <code>TPM</code> datasets can be accessed, such as
<code>TPM_19Q1</code> by EH number.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">eh</span><span class="op">[[</span><span class="st">"EH2264"</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<p>The <code>TPM</code> dataset can also be accessed by using the
<code>depmap_TPM</code> function.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">depmap</span><span class="fu">::</span><span class="fu"><a href="../reference/TPM.html">depmap_TPM</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## see ?depmap and browseVignettes('depmap') for documentation</span></span></code></pre>
<pre><code><span><span class="co">## loading from cache</span></span></code></pre>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 27,024,726 × 6</span></span></span>
<span><span class="co">##    depmap_id  gene          rna_expression entrez_id gene_name cell_line        </span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>            </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> ACH-001113 TSPAN6 (7105)         4.33        <span style="text-decoration: underline;">7</span>105 TSPAN6    LC1SQSF_LUNG     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> ACH-001289 TSPAN6 (7105)         4.57        <span style="text-decoration: underline;">7</span>105 TSPAN6    COGAR359_SOFT_TI…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> ACH-001339 TSPAN6 (7105)         3.15        <span style="text-decoration: underline;">7</span>105 TSPAN6    COLO794_SKIN     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> ACH-001538 TSPAN6 (7105)         5.09        <span style="text-decoration: underline;">7</span>105 TSPAN6    KKU213_BILIARY_T…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> ACH-000242 TSPAN6 (7105)         6.73        <span style="text-decoration: underline;">7</span>105 TSPAN6    RT4_URINARY_TRACT</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> ACH-000708 TSPAN6 (7105)         4.27        <span style="text-decoration: underline;">7</span>105 TSPAN6    SNU283_LARGE_INT…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> ACH-000327 TSPAN6 (7105)         3.34        <span style="text-decoration: underline;">7</span>105 TSPAN6    NCIH1395_LUNG    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> ACH-000233 TSPAN6 (7105)         0.056<span style="text-decoration: underline;">6</span>      <span style="text-decoration: underline;">7</span>105 TSPAN6    DEL_HAEMATOPOIET…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> ACH-000461 TSPAN6 (7105)         4.02        <span style="text-decoration: underline;">7</span>105 TSPAN6    SNU1196_BILIARY_…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> ACH-000705 TSPAN6 (7105)         4.41        <span style="text-decoration: underline;">7</span>105 TSPAN6    LC1F_LUNG        </span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 27,024,716 more rows</span></span></span></code></pre>
</div>
<div class="section level3">
<h3 id="cancer-cell-lines">Cancer cell lines<a class="anchor" aria-label="anchor" href="#cancer-cell-lines"></a>
</h3>
<p>The <code>metadata</code> dataset contains the metadata about all of
the cancer cell lines. It corresponds to the
<code>depmap_19Q1_cell_lines.csv</code> file.</p>
<p>Specific <code>metadata</code> datasets can be accessed, such as
<code>metadata_19Q1</code> by EH number.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">eh</span><span class="op">[[</span><span class="st">"EH2266"</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<p>The most recent <code>metadata</code> dataset can be automatically
loaded into R by using the <code>depmap_metadata</code> function.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">depmap</span><span class="fu">::</span><span class="fu"><a href="../reference/metadata.html">depmap_metadata</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## see ?depmap and browseVignettes('depmap') for documentation</span></span></code></pre>
<pre><code><span><span class="co">## loading from cache</span></span></code></pre>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 1,840 × 29</span></span></span>
<span><span class="co">##    depmap_id  cell_line_name stripped_cell_line_name cell_line aliases cosmic_id</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> ACH-000016 SLR 21         SLR21                   SLR21_KI… <span style="color: #BB0000;">NA</span>             <span style="color: #BB0000;">NA</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> ACH-000032 MHH-CALL-3     MHHCALL3                MHHCALL3… <span style="color: #BB0000;">NA</span>             <span style="color: #BB0000;">NA</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> ACH-000033 NCI-H1819      NCIH1819                NCIH1819… <span style="color: #BB0000;">NA</span>             <span style="color: #BB0000;">NA</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> ACH-000043 Hs 895.T       HS895T                  HS895T_F… <span style="color: #BB0000;">NA</span>             <span style="color: #BB0000;">NA</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> ACH-000049 HEK TE         HEKTE                   HEKTE_KI… <span style="color: #BB0000;">NA</span>             <span style="color: #BB0000;">NA</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> ACH-000051 TE 617.T       TE617T                  TE617T_S… <span style="color: #BB0000;">NA</span>             <span style="color: #BB0000;">NA</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> ACH-000064 SALE           SALE                    SALE_LUNG <span style="color: #BB0000;">NA</span>             <span style="color: #BB0000;">NA</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> ACH-000068 REC-1          REC1                    REC1_HAE… <span style="color: #BB0000;">NA</span>             <span style="color: #BB0000;">NA</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> ACH-000071 <span style="color: #BB0000;">NA</span>             HS706T                  HS706T_B… <span style="color: #BB0000;">NA</span>             <span style="color: #BB0000;">NA</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> ACH-000076 NCO2           NCO2                    NCO2_HAE… <span style="color: #BB0000;">NA</span>             <span style="color: #BB0000;">NA</span></span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 1,830 more rows</span></span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 23 more variables: sex &lt;chr&gt;, source &lt;chr&gt;, RRID &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   WTSI_master_cell_ID &lt;dbl&gt;, sample_collection_site &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   primary_or_metastasis &lt;chr&gt;, primary_disease &lt;chr&gt;, subtype_disease &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   age &lt;chr&gt;, sanger_id &lt;chr&gt;, additional_info &lt;chr&gt;, lineage &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   lineage_subtype &lt;chr&gt;, lineage_sub_subtype &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   lineage_molecular_subtype &lt;chr&gt;, default_growth_pattern &lt;chr&gt;, …</span></span></span></code></pre>
</div>
<div class="section level3">
<h3 id="mutation-calls">Mutation calls<a class="anchor" aria-label="anchor" href="#mutation-calls"></a>
</h3>
<p>The <code>mutationCalls</code> dataset contains all merged mutation
calls (coding region, germline filtered) found in the depmap dependency
study. This dataset corresponds with the
<code>depmap_19Q1_mutation_calls.csv</code> file.</p>
<p>Specific <code>mutationCalls</code> datasets can be accessed, such as
<code>mutationCalls_19Q1</code> by EH number.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">eh</span><span class="op">[[</span><span class="st">"EH2265"</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<p>The most recent <code>mutationCalls</code> dataset can be
automatically loaded into R by using the
<code>depmap_mutationCalls</code> function.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">depmap</span><span class="fu">::</span><span class="fu"><a href="../reference/mutationCalls.html">depmap_mutationCalls</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## see ?depmap and browseVignettes('depmap') for documentation</span></span></code></pre>
<pre><code><span><span class="co">## loading from cache</span></span></code></pre>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 1,235,466 × 32</span></span></span>
<span><span class="co">##    depmap_id  gene_name entrez_id ncbi_build chromosome start_pos end_pos strand</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> ACH-000001 VPS13D        <span style="text-decoration: underline;">55</span>187         37 1           12<span style="text-decoration: underline;">359</span>347  1.24<span style="color: #949494;">e</span>7 +     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> ACH-000001 AADACL4      <span style="text-decoration: underline;">343</span>066         37 1           12<span style="text-decoration: underline;">726</span>308  1.27<span style="color: #949494;">e</span>7 +     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> ACH-000001 IFNLR1       <span style="text-decoration: underline;">163</span>702         37 1           24<span style="text-decoration: underline;">484</span>172  2.45<span style="color: #949494;">e</span>7 +     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> ACH-000001 TMEM57        <span style="text-decoration: underline;">55</span>219         37 1           25<span style="text-decoration: underline;">785</span>018  2.58<span style="color: #949494;">e</span>7 +     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> ACH-000001 ZSCAN20        <span style="text-decoration: underline;">7</span>579         37 1           33<span style="text-decoration: underline;">954</span>141  3.40<span style="color: #949494;">e</span>7 +     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> ACH-000001 POU3F1         <span style="text-decoration: underline;">5</span>453         37 1           38<span style="text-decoration: underline;">512</span>139  3.85<span style="color: #949494;">e</span>7 +     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> ACH-000001 MAST2         <span style="text-decoration: underline;">23</span>139         37 1           46<span style="text-decoration: underline;">498</span>028  4.65<span style="color: #949494;">e</span>7 +     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> ACH-000001 GBP4         <span style="text-decoration: underline;">115</span>361         37 1           89<span style="text-decoration: underline;">657</span>103  8.97<span style="color: #949494;">e</span>7 +     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> ACH-000001 VAV3          <span style="text-decoration: underline;">10</span>451         37 1          108<span style="text-decoration: underline;">247</span>170  1.08<span style="color: #949494;">e</span>8 +     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> ACH-000001 NBPF20    100<span style="text-decoration: underline;">288</span>142         37 1          148<span style="text-decoration: underline;">346</span>689  1.48<span style="color: #949494;">e</span>8 +     </span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 1,235,456 more rows</span></span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 24 more variables: var_class &lt;chr&gt;, var_type &lt;chr&gt;, ref_allele &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   alt_allele &lt;chr&gt;, dbSNP_RS &lt;chr&gt;, dbSNP_val_status &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   genome_change &lt;chr&gt;, annotation_trans &lt;chr&gt;, cDNA_change &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   codon_change &lt;chr&gt;, protein_change &lt;chr&gt;, is_deleterious &lt;lgl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   is_tcga_hotspot &lt;lgl&gt;, tcga_hsCnt &lt;dbl&gt;, is_cosmic_hotspot &lt;lgl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   cosmic_hsCnt &lt;dbl&gt;, ExAC_AF &lt;dbl&gt;, var_annotation &lt;chr&gt;, …</span></span></span></code></pre>
</div>
<div class="section level3">
<h3 id="drug-sensitivity">Drug Sensitivity<a class="anchor" aria-label="anchor" href="#drug-sensitivity"></a>
</h3>
<p>The <code>drug_sensitivity</code> dataset contains dependency data
for cancer cell lines treated with various compounds. This dataset
corresponds with the
<code>primary_replicate_collapsed_logfold_change.csv</code> file.</p>
<p>Specific <code>drug_sensitivity</code> datasets can be accessed, such
as <code>drug_sensitivity_19Q3</code> by EH number.</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span> <span class="va">eh</span><span class="op">[[</span><span class="st">"EH3087"</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<p>The most recent <code>drug_sensitivity</code> dataset can be
automatically loaded into R by using the
<code>depmap_drug_sensitivity</code> function.</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">depmap</span><span class="fu">::</span><span class="fu"><a href="../reference/drug_sensitivity.html">depmap_drug_sensitivity</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## see ?depmap and browseVignettes('depmap') for documentation</span></span></code></pre>
<pre><code><span><span class="co">## loading from cache</span></span></code></pre>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 2,708,508 × 14</span></span></span>
<span><span class="co">##    depmap_id  cell_line compound dependency broad_id name   dose screen_id moa  </span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> ACH-000001 NIHOVCAR… BRD-A00…    -<span style="color: #BB0000;">0.015</span><span style="color: #BB0000; text-decoration: underline;">6</span> BRD-A00… 8-br…   2.5 HTS       PKA …</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> ACH-000007 LS513_LA… BRD-A00…    -<span style="color: #BB0000;">0.095</span><span style="color: #BB0000; text-decoration: underline;">7</span> BRD-A00… 8-br…   2.5 HTS       PKA …</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> ACH-000008 A101D_SK… BRD-A00…     0.379  BRD-A00… 8-br…   2.5 HTS       PKA …</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> ACH-000010 NCIH2077… BRD-A00…     0.119  BRD-A00… 8-br…   2.5 HTS       PKA …</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> ACH-000011 253J_URI… BRD-A00…     0.145  BRD-A00… 8-br…   2.5 HTS       PKA …</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> ACH-000012 HCC827_L… BRD-A00…     0.103  BRD-A00… 8-br…   2.5 HTS       PKA …</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> ACH-000013 ONCODG1_… BRD-A00…     0.353  BRD-A00… 8-br…   2.5 HTS       PKA …</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> ACH-000014 HS294T_S… BRD-A00…     0.128  BRD-A00… 8-br…   2.5 HTS       PKA …</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> ACH-000015 NCIH1581… BRD-A00…     0.167  BRD-A00… 8-br…   2.5 HTS       PKA …</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> ACH-000018 T24_URIN… BRD-A00…     0.832  BRD-A00… 8-br…   2.5 HTS       PKA …</span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 2,708,498 more rows</span></span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 5 more variables: target &lt;chr&gt;, disease_area &lt;chr&gt;, indication &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   smiles &lt;chr&gt;, phase &lt;chr&gt;</span></span></span></code></pre>
</div>
<div class="section level3">
<h3 id="proteomic">Proteomic<a class="anchor" aria-label="anchor" href="#proteomic"></a>
</h3>
<p>The <code>proteomic</code> dataset contains normalized quantitative
profiling of proteins of cancer cell lines by mass spectrometry. This
dataset corresponds with the
<code>https://gygi.med.harvard.edu/sites/gygi.med.harvard.edu/files/documents/protein_quant_current_normalized.csv.gz</code>
file.</p>
<p>Specific <code>proteomic</code> datasets can be accessed, such as
<code>proteomic_20Q2</code> by EH number.</p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">eh</span><span class="op">[[</span><span class="st">"EH3459"</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<p>The most recent <code>proteomic</code> dataset can be automatically
loaded into R by using the <code>depmap_proteomic</code> function.</p>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">depmap</span><span class="fu">::</span><span class="fu"><a href="../reference/proteomic.html">depmap_proteomic</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## see ?depmap and browseVignettes('depmap') for documentation</span></span></code></pre>
<pre><code><span><span class="co">## loading from cache</span></span></code></pre>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 4,821,390 × 12</span></span></span>
<span><span class="co">##    depmap_id  gene_name entrez_id protein    protein_expression protein_id desc </span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> ACH-000849 SLC12A2        <span style="text-decoration: underline;">6</span>558 MDAMB468_…            2.11    sp|P55011… S12A…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> ACH-000441 SLC12A2        <span style="text-decoration: underline;">6</span>558 SH4_SKIN_…            0.070<span style="text-decoration: underline;">5</span>  sp|P55011… S12A…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> ACH-000248 SLC12A2        <span style="text-decoration: underline;">6</span>558 AU565_BRE…           -<span style="color: #BB0000;">0.464</span>   sp|P55011… S12A…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> ACH-000684 SLC12A2        <span style="text-decoration: underline;">6</span>558 KMRC1_KID…           -<span style="color: #BB0000;">0.884</span>   sp|P55011… S12A…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> ACH-000856 SLC12A2        <span style="text-decoration: underline;">6</span>558 CAL51_BRE…            0.789   sp|P55011… S12A…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> ACH-000348 SLC12A2        <span style="text-decoration: underline;">6</span>558 RPMI7951_…           -<span style="color: #BB0000;">0.912</span>   sp|P55011… S12A…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> ACH-000062 SLC12A2        <span style="text-decoration: underline;">6</span>558 RERFLCMS_…            0.729   sp|P55011… S12A…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> ACH-000650 SLC12A2        <span style="text-decoration: underline;">6</span>558 IGR37_SKI…           -<span style="color: #BB0000;">0.658</span>   sp|P55011… S12A…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> ACH-000484 SLC12A2        <span style="text-decoration: underline;">6</span>558 VMRCRCW_K…           -<span style="color: #BB0000;">1.15</span>    sp|P55011… S12A…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> ACH-000625 SLC12A2        <span style="text-decoration: underline;">6</span>558 HEP3B217_…            0.008<span style="text-decoration: underline;">82</span> sp|P55011… S12A…</span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 4,821,380 more rows</span></span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 5 more variables: group_id &lt;dbl&gt;, uniprot &lt;chr&gt;, uniprot_acc &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   TenPx &lt;chr&gt;, cell_line &lt;chr&gt;</span></span></span></code></pre>
</div>
<div class="section level3">
<h3 id="repackaged-data-source">Repackaged data source<a class="anchor" aria-label="anchor" href="#repackaged-data-source"></a>
</h3>
<p>If desired, the original data from which the <a href="https://bioconductor.org/packages/depmap" class="external-link">depmap</a> package were
derived from can be downloaded from the <a href="https://depmap.org/portal/download/" class="external-link">Broad Institute</a> website.
The instructions on how to download these files and how the data was
transformed and loaded into the <a href="https://bioconductor.org/packages/depmap" class="external-link">depmap</a> package can
be found in the <code>make_data.R</code> file found in
<code>./inst/scripts</code>. (It should be noted that the original
uncompressed <em>.csv</em> files are &gt; 1.5GB in total and take a
moderate amount of time to download remotely.)</p>
</div>
</div>
<div class="section level2">
<h2 id="original-depmap-data">Original depmap data<a class="anchor" aria-label="anchor" href="#original-depmap-data"></a>
</h2>
<p>In addition to the re-packaged files, the package also allows to
download any of the original files provided by the <a href="https://figshare.com/authors/Broad_DepMap/5514062" class="external-link">DepMap project
on Figshare</a>.</p>
<p>A list of all the datasets is available with the
<code><a href="../reference/DepMapFiles.html">dmsets()</a></code> function:</p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/DepMapFiles.html">dmsets</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 26 × 5</span></span></span>
<span><span class="co">##    title              dataset_id n_files url                          data      </span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                        <span style="color: #949494; font-style: italic;">&lt;list&gt;</span>    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> DepMap 23Q4 Public   24<span style="text-decoration: underline;">667</span>905      56 https://figshare.com/articl… <span style="color: #949494;">&lt;DpMpDtst&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> DepMap 23Q2 Public   22<span style="text-decoration: underline;">765</span>112      52 https://figshare.com/articl… <span style="color: #949494;">&lt;DpMpDtst&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> DepMap 22Q4 Public   21<span style="text-decoration: underline;">637</span>199      47 https://figshare.com/articl… <span style="color: #949494;">&lt;DpMpDtst&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> DepMap 22Q2 Public   19<span style="text-decoration: underline;">700</span>056      42 https://figshare.com/articl… <span style="color: #949494;">&lt;DpMpDtst&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> DepMap 22Q1 Public   19<span style="text-decoration: underline;">139</span>906      42 https://figshare.com/articl… <span style="color: #949494;">&lt;DpMpDtst&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> DepMap 21Q4 Public   16<span style="text-decoration: underline;">924</span>132      40 https://figshare.com/articl… <span style="color: #949494;">&lt;DpMpDtst&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> DepMap 21Q2 Public   14<span style="text-decoration: underline;">541</span>774      44 https://figshare.com/articl… <span style="color: #949494;">&lt;DpMpDtst&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> DepMap 21Q3 Public   15<span style="text-decoration: underline;">160</span>110      46 https://figshare.com/articl… <span style="color: #949494;">&lt;DpMpDtst&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> DepMap 21Q1 Public   13<span style="text-decoration: underline;">681</span>534      37 https://figshare.com/articl… <span style="color: #949494;">&lt;DpMpDtst&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> DepMap 20Q4 Public   13<span style="text-decoration: underline;">237</span>076      31 https://figshare.com/articl… <span style="color: #949494;">&lt;DpMpDtst&gt;</span></span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 16 more rows</span></span></span></code></pre>
<p>We could check what datasets from any quarter of 2020 are available
by searching for <code>"20Q"</code> in the datasets titles:</p>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/DepMapFiles.html">dmsets</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"20Q"</span>, <span class="va">title</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 4 × 5</span></span></span>
<span><span class="co">##   title                          dataset_id n_files url               data      </span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                               <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;list&gt;</span>    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> DepMap 20Q4 Public               13<span style="text-decoration: underline;">237</span>076      31 https://figshare… <span style="color: #949494;">&lt;DpMpDtst&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> DepMap 20Q2 Public               12<span style="text-decoration: underline;">280</span>541      24 https://figshare… <span style="color: #949494;">&lt;DpMpDtst&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> DepMap 20Q1 Public               11<span style="text-decoration: underline;">791</span>698      24 https://figshare… <span style="color: #949494;">&lt;DpMpDtst&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">4</span> PRISM Repurposing 20Q2 Dataset   20<span style="text-decoration: underline;">564</span>034      16 https://figshare… <span style="color: #949494;">&lt;DpMpDtst&gt;</span></span></span></code></pre>
<p>Let’s focus on the <em>PRISM Repurposing 20Q2 Dataset</em> dataset,
with identifier <code>20564034</code>.</p>
<p>A list of all the files is available with the <code><a href="../reference/DepMapFiles.html">dmfiles()</a></code>
function:</p>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/DepMapFiles.html">dmfiles</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 706 × 7</span></span></span>
<span><span class="co">##    dataset_id       id name                     size download_url md5   mimetype</span></span>
<span><span class="co">##         <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span>   24<span style="text-decoration: underline;">667</span>905 43<span style="text-decoration: underline;">347</span>678 README.txt             2.91<span style="color: #949494;">e</span>4 https://ndo… 4d2d… text/pl…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span>   24<span style="text-decoration: underline;">667</span>905 43<span style="text-decoration: underline;">346</span>361 AchillesCommonEssenti… 1.70<span style="color: #949494;">e</span>4 https://ndo… 1cbf… text/pl…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span>   24<span style="text-decoration: underline;">667</span>905 43<span style="text-decoration: underline;">346</span>367 AchillesHighVarianceG… 7.07<span style="color: #949494;">e</span>3 https://ndo… 3ac0… text/pl…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span>   24<span style="text-decoration: underline;">667</span>905 43<span style="text-decoration: underline;">346</span>370 AchillesNonessentialC… 1.15<span style="color: #949494;">e</span>4 https://ndo… 9b21… text/pl…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span>   24<span style="text-decoration: underline;">667</span>905 43<span style="text-decoration: underline;">346</span>379 AchillesScreenQCRepor… 3.16<span style="color: #949494;">e</span>5 https://ndo… c5dd… text/pl…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span>   24<span style="text-decoration: underline;">667</span>905 43<span style="text-decoration: underline;">346</span>382 AchillesSequenceQCRep… 4.37<span style="color: #949494;">e</span>5 https://ndo… 6ade… text/pl…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span>   24<span style="text-decoration: underline;">667</span>905 43<span style="text-decoration: underline;">346</span>391 AvanaGuideMap.csv      1.60<span style="color: #949494;">e</span>7 https://ndo… b694… text/pl…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span>   24<span style="text-decoration: underline;">667</span>905 43<span style="text-decoration: underline;">346</span>409 AvanaLogfoldChange.csv 3.17<span style="color: #949494;">e</span>9 https://ndo… 58b1… text/pl…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span>   24<span style="text-decoration: underline;">667</span>905 43<span style="text-decoration: underline;">346</span>505 AvanaRawReadcounts.csv 9.70<span style="color: #949494;">e</span>8 https://ndo… a7b5… text/pl…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span>   24<span style="text-decoration: underline;">667</span>905 43<span style="text-decoration: underline;">346</span>574 CRISPRGeneDependency.… 3.94<span style="color: #949494;">e</span>8 https://ndo… b581… text/pl…</span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 696 more rows</span></span></span></code></pre>
<p>If we want to find all files from the <em>PRISM Repurposing 20Q2
Dataset</em> identified above, we could filter all files with its
<code>dataset_id</code>:</p>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/DepMapFiles.html">dmfiles</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">dataset_id</span> <span class="op">==</span> <span class="fl">20564034</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 16 × 7</span></span></span>
<span><span class="co">##    dataset_id       id name                     size download_url md5   mimetype</span></span>
<span><span class="co">##         <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span>   20<span style="text-decoration: underline;">564</span>034 36<span style="text-decoration: underline;">794</span>562 prism-repurposing-20q… 1.92<span style="color: #949494;">e</span>8 https://ndo… d5c2… text/pl…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span>   20<span style="text-decoration: underline;">564</span>034 36<span style="text-decoration: underline;">794</span>565 prism-repurposing-20q… 4.68<span style="color: #949494;">e</span>4 https://ndo… 580b… text/pl…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span>   20<span style="text-decoration: underline;">564</span>034 36<span style="text-decoration: underline;">794</span>571 prism-repurposing-20q… 1.27<span style="color: #949494;">e</span>8 https://ndo… 854c… text/pl…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span>   20<span style="text-decoration: underline;">564</span>034 36<span style="text-decoration: underline;">794</span>574 prism-repurposing-20q… 7.73<span style="color: #949494;">e</span>7 https://ndo… 793d… text/pl…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span>   20<span style="text-decoration: underline;">564</span>034 36<span style="text-decoration: underline;">794</span>577 prism-repurposing-20q… 6.20<span style="color: #949494;">e</span>3 https://ndo… 88f8… text/pl…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span>   20<span style="text-decoration: underline;">564</span>034 36<span style="text-decoration: underline;">794</span>580 prism-repurposing-20q… 4.67<span style="color: #949494;">e</span>7 https://ndo… 8f3e… text/pl…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span>   20<span style="text-decoration: underline;">564</span>034 36<span style="text-decoration: underline;">794</span>583 prism-repurposing-20q… 1.31<span style="color: #949494;">e</span>6 https://ndo… 1e69… text/pl…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span>   20<span style="text-decoration: underline;">564</span>034 36<span style="text-decoration: underline;">794</span>586 prism-repurposing-20q… 5.72<span style="color: #949494;">e</span>6 https://ndo… f1a2… text/pl…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span>   20<span style="text-decoration: underline;">564</span>034 36<span style="text-decoration: underline;">794</span>589 prism-repurposing-20q… 8.47<span style="color: #949494;">e</span>4 https://ndo… 4c3a… text/pl…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span>   20<span style="text-decoration: underline;">564</span>034 36<span style="text-decoration: underline;">794</span>595 prism-repurposing-20q… 2.90<span style="color: #949494;">e</span>8 https://ndo… eb0c… text/pl…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">11</span>   20<span style="text-decoration: underline;">564</span>034 36<span style="text-decoration: underline;">794</span>607 prism-repurposing-20q… 3.08<span style="color: #949494;">e</span>8 https://ndo… 5140… text/pl…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">12</span>   20<span style="text-decoration: underline;">564</span>034 36<span style="text-decoration: underline;">794</span>610 prism-repurposing-20q… 3.32<span style="color: #949494;">e</span>5 https://ndo… 8a48… text/pl…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">13</span>   20<span style="text-decoration: underline;">564</span>034 36<span style="text-decoration: underline;">794</span>613 prism-repurposing-20q… 8.05<span style="color: #949494;">e</span>3 https://ndo… 4741… text/pl…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">14</span>   20<span style="text-decoration: underline;">564</span>034 36<span style="text-decoration: underline;">794</span>616 prism-repurposing-20q… 1.15<span style="color: #949494;">e</span>8 https://ndo… d89c… text/pl…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">15</span>   20<span style="text-decoration: underline;">564</span>034 36<span style="text-decoration: underline;">794</span>619 prism-repurposing-20q… 4.28<span style="color: #949494;">e</span>6 https://ndo… 162d… text/pl…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">16</span>   20<span style="text-decoration: underline;">564</span>034 36<span style="text-decoration: underline;">794</span>622 prism-repurposing-20q… 1.61<span style="color: #949494;">e</span>7 https://ndo… 610f… text/pl…</span></span></code></pre>
<p>Let’s now focus on the
<code>prism-repurposing-20q2-primary-screen-cell-line-info.csv</code>
file. We can filter it by its name and downloaded it with
<code><a href="../reference/DepMapFiles.html">dmget()</a></code>:</p>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/DepMapFiles.html">dmfiles</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">name</span> <span class="op">==</span> <span class="st">"prism-repurposing-20q2-primary-screen-cell-line-info.csv"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/DepMapFiles.html">dmget</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## File: prism-repurposing-20q2-primary-screen-cell-line-info.csv (36794565)</span></span></code></pre>
<pre><code><span><span class="co">## Loading 36794565 from cache.</span></span></code></pre>
<pre><code><span><span class="co">##                                              BFC1 </span></span>
<span><span class="co">## "/github/home/.cache/R/depmap/5cb6aef5e_36794565"</span></span></code></pre>
<p>The <code><a href="../reference/DepMapFiles.html">dmget()</a></code> function will first check if it hasn’t
already been downloaded and cached in the depmap cache directory (see
<code>?dmCache()</code>). If so, it will retrieve if from there.
Otherwise, it will download the file and store it in the package cache
directory. It will return the location of the cached file.</p>
<p>Given that the file is in csv format, we can directly open it with
<code><a href="https://readr.tidyverse.org/reference/read_delim.html" class="external-link">read_csv()</a></code>:</p>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/DepMapFiles.html">dmfiles</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">name</span> <span class="op">==</span> <span class="st">"prism-repurposing-20q2-primary-screen-cell-line-info.csv"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/DepMapFiles.html">dmget</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html" class="external-link">read_csv</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## File: prism-repurposing-20q2-primary-screen-cell-line-info.csv (36794565)</span></span></code></pre>
<pre><code><span><span class="co">## Loading 36794565 from cache.</span></span></code></pre>
<pre><code><span><span class="co">## <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">588</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">10</span></span></span>
<span><span class="co">## <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">## <span style="font-weight: bold;">Delimiter:</span> ","</span></span>
<span><span class="co">## <span style="color: #BB0000;">chr</span> (9): row_name, depmap_id, ccle_name, primary_tissue, secondary_tissue, t...</span></span>
<span><span class="co">## <span style="color: #BBBB00;">lgl</span> (1): passed_str_profiling</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span></span></code></pre>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 588 × 10</span></span></span>
<span><span class="co">##    row_name  depmap_id ccle_name primary_tissue secondary_tissue tertiary_tissue</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> ACH-0008… ACH-0008… KYSE510_… esophagus      esophagus_squam… <span style="color: #BB0000;">NA</span>             </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> ACH-0009… ACH-0009… HEC1A_EN… uterus         uterus_endometr… <span style="color: #BB0000;">NA</span>             </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> ACH-0006… ACH-0006… MIAPACA2… pancreas       <span style="color: #BB0000;">NA</span>               <span style="color: #BB0000;">NA</span>             </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> ACH-0006… ACH-0006… SW620_LA… colorectal     <span style="color: #BB0000;">NA</span>               <span style="color: #BB0000;">NA</span>             </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> ACH-0003… ACH-0003… SKHEP1_L… liver          <span style="color: #BB0000;">NA</span>               <span style="color: #BB0000;">NA</span>             </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> ACH-0008… ACH-0008… SW480_LA… colorectal     <span style="color: #BB0000;">NA</span>               <span style="color: #BB0000;">NA</span>             </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> ACH-0005… ACH-0005… YKG1_CEN… central_nervo… glioma           glioblastoma   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> ACH-0003… ACH-0003… J82_URIN… urinary_tract  <span style="color: #BB0000;">NA</span>               <span style="color: #BB0000;">NA</span>             </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> ACH-0007… ACH-0007… KYSE30_O… esophagus      esophagus_squam… <span style="color: #BB0000;">NA</span>             </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> ACH-0009… ACH-0009… NCIH2172… lung           lung_NSC         <span style="color: #BB0000;">NA</span>             </span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 578 more rows</span></span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 4 more variables: passed_str_profiling &lt;lgl&gt;, str_profiling_notes &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   originally_assigned_depmap_id &lt;chr&gt;, originally_assigned_ccle_name &lt;chr&gt;</span></span></span></code></pre>
<p>It is also possible to pass multiple rows of the
<code><a href="../reference/DepMapFiles.html">dmfiles()</a></code> table to <code><a href="../reference/DepMapFiles.html">dmget()</a></code> to retrieve
multiple file paths. Below, let’s get all the README.txt files from
2020:</p>
<div class="sourceCode" id="cb67"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ids_2020</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="../reference/DepMapFiles.html">dmsets</a></span><span class="op">(</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"20Q"</span>, <span class="va">title</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">dataset_id</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/DepMapFiles.html">dmfiles</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">dataset_id</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">ids_2020</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"README"</span>, <span class="va">name</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/DepMapFiles.html">dmget</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## File: README (25797005)</span></span></code></pre>
<pre><code><span><span class="co">## Loading 25797005 from cache.</span></span></code></pre>
<pre><code><span><span class="co">## File: README (25817903)</span></span></code></pre>
<pre><code><span><span class="co">## Loading 25817903 from cache.</span></span></code></pre>
<pre><code><span><span class="co">## File: README (22897970)</span></span></code></pre>
<pre><code><span><span class="co">## Loading 22897970 from cache.</span></span></code></pre>
<pre><code><span><span class="co">## File: README (22543694)</span></span></code></pre>
<pre><code><span><span class="co">## Loading 22543694 from cache.</span></span></code></pre>
<pre><code><span><span class="co">##                                                BFC2 </span></span>
<span><span class="co">##  "/github/home/.cache/R/depmap/5cbd26e65b_25797005" </span></span>
<span><span class="co">##                                                BFC3 </span></span>
<span><span class="co">## "/github/home/.cache/R/depmap/5cb6ae6eb67_25817903" </span></span>
<span><span class="co">##                                                BFC4 </span></span>
<span><span class="co">## "/github/home/.cache/R/depmap/5cb590891ea_22897970" </span></span>
<span><span class="co">##                                                BFC5 </span></span>
<span><span class="co">##  "/github/home/.cache/R/depmap/5cb1768cb9_22543694"</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<pre><code><span><span class="co">## R Under development (unstable) (2024-03-13 r86113)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.4 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] ExperimentHub_2.11.1 AnnotationHub_3.11.3 BiocFileCache_2.11.1</span></span>
<span><span class="co">##  [4] dbplyr_2.5.0         BiocGenerics_0.49.1  depmap_1.17.5       </span></span>
<span><span class="co">##  [7] lubridate_1.9.3      forcats_1.0.0        stringr_1.5.1       </span></span>
<span><span class="co">## [10] dplyr_1.1.4          purrr_1.0.2          readr_2.1.5         </span></span>
<span><span class="co">## [13] tidyr_1.3.1          tibble_3.2.1         ggplot2_3.5.0       </span></span>
<span><span class="co">## [16] tidyverse_2.0.0      BiocStyle_2.31.0    </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] tidyselect_1.2.1        blob_1.2.4              bitops_1.0-7           </span></span>
<span><span class="co">##  [4] filelock_1.0.3          Biostrings_2.71.4       fastmap_1.1.1          </span></span>
<span><span class="co">##  [7] RCurl_1.98-1.14         digest_0.6.35           mime_0.12              </span></span>
<span><span class="co">## [10] timechange_0.3.0        lifecycle_1.0.4         KEGGREST_1.43.0        </span></span>
<span><span class="co">## [13] RSQLite_2.3.5           magrittr_2.0.3          compiler_4.4.0         </span></span>
<span><span class="co">## [16] rlang_1.1.3             sass_0.4.9              tools_4.4.0            </span></span>
<span><span class="co">## [19] utf8_1.2.4              yaml_2.3.8              knitr_1.45             </span></span>
<span><span class="co">## [22] htmlwidgets_1.6.4       bit_4.0.5               curl_5.2.1             </span></span>
<span><span class="co">## [25] withr_3.0.0             desc_1.4.3              grid_4.4.0             </span></span>
<span><span class="co">## [28] stats4_4.4.0            fansi_1.0.6             colorspace_2.1-0       </span></span>
<span><span class="co">## [31] scales_1.3.0            cli_3.6.2               rmarkdown_2.26         </span></span>
<span><span class="co">## [34] crayon_1.5.2            ragg_1.3.0              generics_0.1.3         </span></span>
<span><span class="co">## [37] httr_1.4.7              tzdb_0.4.0              DBI_1.2.2              </span></span>
<span><span class="co">## [40] cachem_1.0.8            zlibbioc_1.49.3         parallel_4.4.0         </span></span>
<span><span class="co">## [43] AnnotationDbi_1.65.2    BiocManager_1.30.22     XVector_0.43.1         </span></span>
<span><span class="co">## [46] vctrs_0.6.5             jsonlite_1.8.8          bookdown_0.38          </span></span>
<span><span class="co">## [49] IRanges_2.37.1          hms_1.1.3               S4Vectors_0.41.4       </span></span>
<span><span class="co">## [52] bit64_4.0.5             systemfonts_1.0.6       jquerylib_0.1.4        </span></span>
<span><span class="co">## [55] glue_1.7.0              pkgdown_2.0.7.9000      stringi_1.8.3          </span></span>
<span><span class="co">## [58] gtable_0.3.4            BiocVersion_3.19.1      GenomeInfoDb_1.39.9    </span></span>
<span><span class="co">## [61] munsell_0.5.0           pillar_1.9.0            rappdirs_0.3.3         </span></span>
<span><span class="co">## [64] htmltools_0.5.7         GenomeInfoDbData_1.2.11 httr2_1.0.0            </span></span>
<span><span class="co">## [67] R6_2.5.1                textshaping_0.3.7       vroom_1.6.5            </span></span>
<span><span class="co">## [70] evaluate_0.23           Biobase_2.63.0          png_0.1-8              </span></span>
<span><span class="co">## [73] memoise_2.0.1           bslib_0.6.1             xfun_0.42              </span></span>
<span><span class="co">## [76] fs_1.6.3                pkgconfig_2.0.3</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Laurent Gatto, Theo Killian.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.9000.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
